lib/xnvme_be_linux_hugepage.c

```diff
@@ -0,0 +1,179 @@
+// Copyright (C) Simon A. F. Lund <simon.lund@samsung.com>
+// SPDX-License-Identifier: Apache-2.0
+#include <xnvme_be.h>
+#include <xnvme_be_nosys.h>
+#ifdef XNVME_BE_LINUX_ENABLED
+#include <xnvme_dev.h>
+#include <errno.h>
+
+#include <fcntl.h>
+#include <sys/mman.h>
+#include <unistd.h>
+#include <stdio.h>
+#include <sys/queue.h>
+#include <linux/limits.h>
+
+struct huge_alloc {
+	char path[PATH_MAX];
+	int fd;
+	void *addr;
+	size_t len;
+
+	SLIST_ENTRY(huge_alloc) link;
+};
+
+SLIST_HEAD(huge_alloc_slist, huge_alloc);
+struct huge_alloc_slist huge_alloc_head;
+
+size_t
+get_hugepage_size()
+{
+	char line[128];
```

Comment:
It might be good form to initialize the entire string with the nul-terminating char, e.g.:
```
char line[128] = { '\0' };
```